api/graphite_req.go

```diff
@@ -0,0 +1,160 @@
+package api
+
+import (
+	"fmt"
+
+	"github.com/grafana/metrictank/api/models"
+)
+
+// ReqMap is a map of requests of data,
+// it has single requests for which no pre-normalization effort will be performed, and
+// requests are that can be pre-normalized together to the same resolution, bundled by their PNGroup
+type ReqMap struct {
+	single   []models.Req
+	pngroups map[models.PNGroup][]models.Req
+	cnt      uint32
+}
+
+func NewReqMap() *ReqMap {
+	return &ReqMap{
+		pngroups: make(map[models.PNGroup][]models.Req),
+	}
+}
+
```

Comment:
```suggestion
// Add adds a models.Req to the ReqMap in the appropriate member slice based on PNGroup and incrememtns the ReqMap's cnt
```

Reply:
I don't think comments should describe that they put "things into the correct datastructure" internally. The types (ReqsPlan and ReqMap) already describe how they categorize requests.